--- /usr/lib/python2.6/site-packages/salt/cloud/__init__.py	2015-05-11 16:36:32.000000000 -0400
+++ /usr/lib/python2.6/site-packages/salt/cloud/__init__.py	2015-06-03 21:30:25.000000000 -0400
@@ -15,6 +15,7 @@
 import time
 import signal
 import logging
+import shutil
 import multiprocessing
 from itertools import groupby
 from salt.ext.six.moves import input
@@ -1028,6 +1029,13 @@
             minion_dict = salt.config.get_cloud_config_value(
                 'minion', vm_, self.opts, default={}
             )
+
+            ##CUSTOM CODE##
+            m_cache = os.path.join(self.opts['cachedir'], "minions", minion_dict.get('id', name))
+            if os.path.isdir(m_cache):
+                shutil.rmtree(m_cache)
+
+
             key_file = os.path.join(
                 self.opts['pki_dir'], 'minions', minion_dict.get('id', name)
             )
@@ -1041,6 +1049,7 @@
                     )
                 continue
 
+
             if os.path.isfile(key_file) and not globbed_key_file:
                 # Single key entry. Remove it!
                 salt.utils.cloud.remove_key(self.opts['pki_dir'], os.path.basename(key_file))
